x-pack/plugins/security_solution/public/sourcerer/experimental/containers/dataview_picker_provider.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useEffect, type FC, type PropsWithChildren } from 'react';
+import { Provider as ReduxProvider } from 'react-redux';
+
+import { useDispatch } from '../redux/hooks';
+import { Context } from '../redux/context';
+import { store, listenerMiddleware, startAppListening } from '../redux/store';
+import { useKibana } from '../../../common/lib/kibana';
+import { createChangeDataviewListener, createInitDataviewListener } from '../redux/listeners';
+import { init } from '../redux/actions';
+import { DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID } from '../constants';
+
+// NOTE: this can be spawned multiple times, eq. when you need something like a separate data view picker for a subsection of the app -
+// for example, in the timeline.
+export const DataviewPickerProvider: FC<PropsWithChildren<{}>> = memo(({ children }) => {
```

Comment:
Is this the primary reason for a separate provider? Wanting to keep the listeners to their own store?

Reply:
yeah, + being able to do multiple instances of the component... though I decided to remove it. the provider is there but it will just kick off the loading for now. will probably be moved somehere else.